crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -1108,15 +1110,35 @@ impl SsaContext {
         NodeId(index)
     }
 
-    pub fn get_builtin_opcode(&self, node_id: NodeId) -> Option<builtin::Opcode> {
+    pub fn get_builtin_opcode(
+        &self,
+        node_id: NodeId,
+        arguments: &[Expression],
+    ) -> Option<builtin::Opcode> {
         match &self[node_id] {
-            NodeObject::Function(FunctionKind::Builtin(opcode), ..) => Some(*opcode),
+            NodeObject::Function(FunctionKind::Builtin(opcode), ..) => match opcode {
+                builtin::Opcode::Println(_) => {
+                    assert!(
+                        arguments.len() == 1,
+                        "print statements currently only support one argument"
```

Comment:
Shouldn't this be verified by the typechecker? Or I suppose this case may be reachable if a struct was passed in that was expanded into multiple arguments? We should reference "no structs/tuple" (or functions?) explicitly instead if that is the case.

Reply:
You are correct this will be caught by the typechecker and I added it more as a sanity check. A struct ident at this point will still be seen as a singular argument. An expanded struct will hit the assert in the first line of `evaluate_println`. However, the assert in `evaluate_println` is not explicit about this either and I will make the error for structs/tuples more clear.

Reply:
The error for structs/tuples here is now more explicit